pycbc/transforms.py

```diff
@@ -592,26 +596,133 @@ def inverse_transform(self, maps):
+        out[parameters.chirp_distance] = conversions.chirp_distance(
+            maps[parameters.distance], maps[parameters.mchirp], ref_mass=self.ref_mass
+        )
         return self.format_output(maps, out)
 
     def jacobian(self, maps):
         """Returns the Jacobian for transforming chirp distance to
         luminosity distance, given the chirp mass.
         """
-        ref_mass=1.4
-        mchirp = maps['mchirp']
-        return (2.**(-1./5) * self.ref_mass / mchirp)**(-5./6)
+        ref_mass = 1.4
+        mchirp = maps["mchirp"]
+        return (2.0 ** (-1.0 / 5) * self.ref_mass / mchirp) ** (-5.0 / 6)
 
     def inverse_jacobian(self, maps):
         """Returns the Jacobian for transforming luminosity distance to
         chirp distance, given the chirp mass.
         """
-        ref_mass=1.4
-        mchirp = maps['mchirp']
-        return (2.**(-1./5) * self.ref_mass / mchirp)**(5./6)
+        ref_mass = 1.4
+        mchirp = maps["mchirp"]
+        return (2.0 ** (-1.0 / 5) * self.ref_mass / mchirp) ** (5.0 / 6)
+
+
+class AlignTotalSpin(BaseTransform):
+
+    """Parameters
+    ----------
+    iota : str
+        The name of the iota parameter
+    s1 : str
+        The name of the s1 parameter
+    s2 : str
+        The name of the s2 parameter
+    m1 : str
+        The name of the m1 parameter
+    m2 : str
+        The name of the m2 parameter
+    f_lower : str
+        The name of the f_lower parameter
+    phi_ref : str
+        The name of the phi_ref parameter
+    """
+
+    name = "align_total_spin"
+
+    def __init__(self, iota, s1, s2, m1, m2, f_lower, phi_ref, newIota, news1, news2):
+        self.iota = iota
+        self.s1 = s1
+        self.s2 = s2
+        self.m1 = m1
+        self.m2 = m2
+        self.f_lower = f_lower
+        self.phi_ref = phi_ref
+        self.newIota = newIota
+        self.news1 = news1
+        self.news2 = news2
+        self._inputs = [
+            self.iota,
+            self.s1,
+            self.s2,
+            self.m1,
+            self.m2,
+            self.f_lower,
+            self.phi_ref,
+        ]
+        self._outputs = [self.newIota, self.news1, self.news2]
+        super(AlignTotalSpin, self).__init__()
+
+    def transform(self, maps):
+        """
+        Rigidly rotate binary so that the total angular momentum has the given
+        inclination (iota) instead of the orbital angular momentum. Return
+        the new inclination, s1, and s2. s1 and s2 are dimensionless spin.
+        Note: the spins are assumed to be given in the frame defined by the orbital
+        angular momentum.
+        """
+
+        iota = self.iota
+        s1 = self.s1
+        s2 = self.s2
+        m1 = self.m1
+        m2 = self.m2
+        f_lower = self.f_lower
+        phi_ref = self.phi_ref
+
+        if numpy.linalg.norm(maps[s1][:2]) != 0 or numpy.linalg.norm(maps[s1][:2]) != 0:
```

Comment:
This can be replaced without calling the numpy function
```suggestion
        if not all(s == 0.0 for s in s1[:2]) or not all(s == 0.0 for s in s2[:2]):
```